cache/metrics.go

```diff
@@ -34,6 +35,10 @@ type MetricsCache struct {
 	deleteCount counter
 }
 
+func (c MetricsCache) Scan(ctx context.Context, key string) (map[string]string, error) {
+	return c.next.Scan(ctx, key)
```

Comment:
Should we also track some metrics here for the scan call like we do for the other read operations? https://github.com/harness/ff-proxy/pull/207/files#diff-1d62a698b9e6a5619c58e046545dc316861cbddfa80279c049c70a55196c9ef2R115

Would probably just need to add this in. Or maybe it'd make more sense to add a specific prometheus metric for `scan` to separate it from the Get calls?
```go
	defer func() {
		trackHistogram(start, c.readDuration)
		trackCounter(c.readCount, key, getErrorLabel(err))
	}()
```